src/virtual-source-buffer.js

```diff
@@ -247,29 +241,33 @@ export default class VirtualSourceBuffer extends videojs.EventTarget {
     let types = ['audio', 'video'];
 
     types.forEach((type) => {
+      // Don't create a SourceBuffer of this type if we don't have a
+      // codec for it
       if (!this[`${type}Codec_`]) {
         return;
       }
 
-      // if the mediasource already has a buffer for the codec
-      // use that
-      if (this.mediaSource_[`${type}Buffer_`]) {
-        this[`${type}Buffer_`] = this.mediaSource_[`${type}Buffer_`];
+      // Do nothing if a SourceBuffer of this type already exists
+      if (this[`${type}Buffer_`]) {
         return;
       }
 
-      let buffer = this.nativeMediaSource.addSourceBuffer(
-        type + '/mp4;codecs="' + this[`${type}Codec_`] + '"'
-      );
+      let buffer = null;
 
-      this.mediaSource_[`${type}Buffer_`] = buffer;
-      this[`${type}Buffer_`] = buffer;
-      // we only add audio event handlers here if we also have
-      // a videoCodec. we will add seprate event handlers
-      // for audio only after this block of code
-      if (!this.videoCodec_) {
-        return;
+      // If the mediasource already has a SourceBuffer for the codec
+      // use that
+      if (this.mediaSource_[`${type}Buffer_`]) {
+        buffer = this.mediaSource_[`${type}Buffer_`];
+      } else {
+        buffer = this.nativeMediaSource.addSourceBuffer(
```

Comment:
`this.mediaSource_` is our HtmlMediaSource and `this.nativeMediaSource` is the native, real one. Having two separate properties is much easier to understand that doing `this.mediaSource_.mediaSource_` to get to the native implementation.

I'm going to add an underscore to nativeMediaSource since it shouldn't be used outside of this module.